main.py

```diff
@@ -515,72 +527,36 @@ def main(cl_arguments):
-                assert layer_path, "No best checkpoint found."
-                load_model_state(
-                    model,
-                    layer_path,
-                    args.cuda,
-                    strict=strict,
-                    skip_task_models=task_names_to_avoid_loading,
-                )
-
     if args.do_full_eval:
-        # Evaluate
         log.info("Evaluating...")
         splits_to_write = evaluate.parse_write_preds_arg(args.write_preds)
-        if args.transfer_paradigm == "finetune":
-            if args.do_target_task_training:
-                for task in target_tasks:
-                    ckpt_path = get_best_checkpoint_path(args.run_dir, "target_train", task.name)
-                    assert ckpt_path is not None and ".best" in ckpt_path
-                    load_model_state(
-                        model, ckpt_path, args.cuda, skip_task_models=[], strict=strict
-                    )
-
-                    evaluate_and_write(args, model, [task], splits_to_write)
-            elif args.do_pretrain:
-                # If args.do_target_task_training = 0 and args.do_pretrain = 1
-                # then evaluate on pretraining checkpoints.
-                for task in pretrain_tasks:
-                    ckpt_path = get_best_checkpoint_path(args.run_dir, "pretrain", task.name)
-                    assert ckpt_path is not None and ".best" in ckpt_path
-                    load_model_state(
-                        model, ckpt_path, args.cuda, skip_task_models=[], strict=strict
-                    )
-                    evaluate_and_write(args, model, [task], splits_to_write)
-
-        elif args.transfer_paradigm == "frozen":
-            # Don't do any special checkpointing logic here
-            # since model already has all the trained task specific modules.
-            evaluate_and_write(args, model, target_tasks, splits_to_write)
+        if args.do_target_task_training or (args.load_model and not args.do_pretrain):
+            # If we either do target task training, or if we only evaluate
+            # without pretraining or target task training
+            # then we evaluate on the target tasks.
+            for task in target_tasks:
+                # Find the task-specific best checkpoint to evaluate on.
+                ckpt_path = get_best_checkpoint_path(args, "eval", task.name)
+                assert ckpt_path is not None and ".best" in ckpt_path
+                load_model_state(model, ckpt_path, args.cuda, skip_task_models=[], strict=strict)
+
+                evaluate_and_write(args, model, [task], splits_to_write)
+        elif args.do_pretrain:
+            # If args.do_target_task_training = 0 and args.do_pretrain = 1
+            # then evaluate on pretraining checkpoints.
+            for task in pretrain_tasks:
+                ckpt_path = get_best_checkpoint_path(args, "target_train", task.name)
```

Comment:
I think it's worth revising the arguments to the get_best_checkpoint_path method, both for clarity and to fix a bug: Passing in the wrong phase name feels like a hack, and as-is, this means that if you turn off do_target_task_training, then load_eval_checkpoint will stop working.

Reply:
What do you mean by if do_target_task_training is off, then load_eval_checkpoint will not work? load_eval_checpoint is checked in a different logic branch in get_best_checkpoint than target training.

Reply:
I actually realize you don't have to call get_best_checkpoint here for simplicity.